holmes/plugins/toolsets/azuremonitor_metrics/utils.py

```
@@ -0,0 +1,714 @@
+                for resource in resources:
+                    # Return the first AKS cluster found
+                    return resource.id
+                    
+    except Exception as e:
+        logging.debug(f"Failed to get AKS cluster resource ID from metadata: {e}")
+    
+    try:
+        # Fallback: Try to get cluster info from Kubernetes environment
+        # Check if we're running in a Kubernetes pod with service account
+        with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r") as f:
+            namespace = f.read().strip()
```

Ruff (0.12.2)

Local variable `namespace` is assigned to but never used

Remove assignment to unused variable `namespace`

(F841)